app/features/project-badges.tsx

```diff
@@ -1,17 +1,39 @@
 import type { Project } from "@prisma/client";
+import { useState } from "react";
 import { Badge, ProjectAvatar } from "~/components";
 
 export function ProjectBadges({ projects }: { projects: Project[] }) {
+  const [allIcons, setAllIcons] = useState<boolean>(false);
+  const max = 3;
```

Comment:
could move this outside the component at the top to make it more obvious that it's configuration
```CONST MAX = 3```